src/test/java/org/mockitousage/annotation/SpyAnnotationTest.java

```diff
@@ -173,18 +192,86 @@ public void should_report_when_encosing_instance_is_needed() throws Exception {
         }
     }
 
-    static class NestedClassWithoutDefinedConstructor { }
+    @Test
+    public void should_report_private_inner_not_supported() throws Exception {
+        try {
+            MockitoAnnotations.initMocks(new WithInnerPrivate());
+            fail();
+        } catch (MockitoException e) {
+            // Currently fails at instantiation time, because the mock subclass don't have the
+            // 1-arg constructor expected for the outerclass.
+            // org.mockito.internal.creation.instance.ConstructorInstantiator.withParams()
+            assertThat(e).hasMessageContaining("Unable to initialize @Spy annotated field 'spy_field'")
```

Comment:
@raphw I found a slight edge issue here, when working on inner private. In this case, the created mock class loses it's 1-arg constructor, the error is raised in the above method `ConstructorInstantiator.withParams`.

I feel this is either a bug or a misconfiguration. But I haven't looked too much at it yet. Could it be the combination of modifiers that causes this problem ?

The class to mock is [`InnerPrivate`](https://github.com/mockito/mockito/pull/885/files#diff-3404c0c83dc4764960b54a85a537418fR252).